Observer/Product/AbstractChangedProductObserver.php

```diff
@@ -104,9 +114,19 @@ public function execute(Observer $observer)
      */
     protected function registerChangedItemStore(ProductInterface $product, int $storeId)
     {
-        $changedItem = $this->createChangedItem($product, $storeId);
-        if (!$this->changedItemRepository->exists($changedItem)) {
-            $this->changedItemRepository->save($changedItem);
+        $itemId = $product->getId();
+
+        $itemsToInsert = [$itemId];
+        $parentProduct = $this->configurableProductType->getParentIdsByChild($itemId);
+        if (count($parentProduct) > 0) {
+            $itemsToInsert = $parentProduct;
+        }
```

Comment:
If has parents, we're not inserting the item?
Shouldn't we do an append?

Reply:
Exactly, the issue here is that when a user edit a variant product now we're storing only the variant but not the parent, so the request is send to dooplugins and when we are going to manage the variant it is taken as a `simple` product instead of a variant and it is managed as an simple product, which mean that it is exclueded of the configurable product and if the customer has the feed configured as `group variants` now the variants appear in the search... It is because for magento a variant is a simple product but included in a group, it is quite weired, so to avoid this case when a variant product is updated we store its parent to be updated, and in dooplugins when we update this product we update it and its variants, so the product that was edited will be updated in the feed also.

Reply:
makes sense then!
Maybe we could add a small comment alongside these lines to leave evidence of this behavior
```php
/* When updating a product it's children are also updated, so in this case we don't need to specify the itemId */
```

Reply:
great! thank you :)